test/testsuits.h

```diff
@@ -0,0 +1,49 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#pragma once
+
+#include <test/TestCase.h>
+#include <test/libsolidity/ASTJSONTest.h>
+#include <test/libsolidity/SyntaxTest.h>
+#include <test/libsolidity/SMTCheckerJSONTest.h>
+#include <test/libyul/YulOptimizerTest.h>
+
+#include <boost/filesystem.hpp>
+
+using namespace dev::solidity::test;
+
+/** Container for all information regarding a testsuite */
+struct Testsuite
+{
+	const char* title;
+	const boost::filesystem::path path;
+	const boost::filesystem::path subpath;
+	bool smt;
+	bool ipc;
+	TestCase::TestCaseCreator testCaseCreator;
+};
+
+
+/// Array of all testsuits
```

Comment:
This is just the list of test suites that are usable with `isoltest`. I think the name of this file should also reflect that. Perhaps `InteractiveTests.h`?

Reply:
``InteractiveTests.h`` sounds more like the tests are *only* to be run interactively.

It's all extracted tests, isn't it? So if we want to be more specific than just "testsuits", then maybe that's the way to go, although ``ExtractedTests.h`` doesn't sound good either IMO, so not sure, maybe there's something else that's better...

Reply:
Another thing they have in common is that they generate expectations that are compared. They are also auto-updatable. Just some inspiration :)

Reply:
"extracted" implies that they are not in their "normal" place but put there, taken from their normal environment. That seems neither correct nor important in this context? 

Why are you unhappy with `testsuites.h`?

Reply:
`testsuites.h` is too generic, because not all test suites can be covered by this.

Reply:
Maybe ``TestCaseDirectories``? I'm terrible at naming things ;-). But since the base class for the extracted tests is named ``TestCase`` anyways (although it's again not *all* of the test cases...)...